subiquitycore/core.py

```diff
@@ -475,10 +479,21 @@ def run(self):
                 self.run_scripts(self.common['opts'].scripts)
             controllers_mod = __import__('%s.controllers' % self.project,
                                          None, None, [''])
-            for k in self.controllers:
-                log.debug("Importing controller: {}".format(k))
-                klass = getattr(controllers_mod, k+"Controller")
-                self.common['controllers'][k] = klass(self.common)
+            for i, k in enumerate(self.controllers):
```

Comment:
Is this changing because subiquity isn't restarting to get the new controllers, or because the refresh could cause duplicates to exist (how come? seems like the new rev should /replace/ everything)?

Or is this because the same controller could get shown twice; ie. ordering could change across a refresh?

Reply:
Yeah this is about the refresh controller appearing more than once in the list. If the check shows there is an update available in time, it offers it after language selection. If it didn't offer an update then, it also asks again after mirror selection.